storage2/src/snapshot.rs

```diff
@@ -28,3 +37,50 @@ impl Snapshot {
         self.jmt_version
     }
 }
+
+/// A reader interface for rocksdb. NOTE: it is up to the caller to ensure consistency between the
+/// rocksdb::DB handle and any write batches that may be applied through the writer interface.
+impl TreeReader for Snapshot {
+    /// Gets node given a node key. Returns `None` if the node does not exist.
+    fn get_node_option(&self, node_key: &NodeKey) -> Result<Option<Node>> {
+        let node_key = node_key.clone();
+
+        let span = Span::current();
+
+        let jmt_cf = self
+            .db
+            .cf_handle("jmt")
+            .expect("jmt column family not found");
+        let value = self
+            .rocksdb_snapshot
+            .get_cf(jmt_cf, &node_key.encode()?)?
+            .map(|db_slice| Node::decode(&db_slice))
+            .transpose()?;
+
+        tracing::trace!(?node_key, ?value);
```

Comment:
Do we want to trace after the possible errors could be thrown, or before? I think before, so that the trace shows up even if an error is thrown.